.github/workflows/ci-conan.yml

```diff
@@ -179,11 +179,14 @@ jobs:
           cache-name: cache-conan-data
         with:
           path: ~/.conan/data
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/metadata.json') }}
+          key: build-${{ matrix.config.os }}-${{ matrix.build_type }}-${{ matrix.config.compiler.type }}-${{ matrix.config.compiler.version }}-${{ matrix.config.lib }}
           restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
+            build-${{ matrix.config.os }}-${{ matrix.build_type }}-${{ matrix.config.compiler.type }}-${{ matrix.config.compiler.version }}-${{ matrix.config.lib }}
```

Comment:
This is a short-form version of the information in the resulting `conaninfo.txt` file that is written into the build directory.  Ideally, we could just use:
`key: build-${{ env.cache-name }}-${{ hashFiles('**/conaninfo.json') }}`

This would ensure build settings were unique (i.e. os, configuration, compiler, architecture and setting).  However, this file is not written until `conan install` is called, but would be required to be present before hand in order to sync the cache.  I can't see any way to access this, I don't think a dry-run of `conan install` produces this...

Not sure if you can see any other alternatives here?